### PR TITLE
Write translation guide

### LINKED ISSUE #1464: Translation guide for translators
We only have English :us: and Japanese :jp: translations. We should have a guide explaining how to provide new translations for the UI geared towards translators.

### PR DESCRIPTION
**Pull request by maciaszczykm**
Closes #1464.

### CODE REVIEW

**Comment by cheld** [comment]
Wow. Can you link to our existing resources:

https://github.com/kubernetes/dashboard/blob/master/docs/devel/text-conventions.md

https://github.com/kubernetes/dashboard/blob/master/docs/devel/localization.md

**Comment by cheld** [comment]
What about some words of warning to developers. Something like, "if you refactor ... keep in mind .... that this and that action... destroys existing translation...in this and that situation." Just in case we have situations like this....

**Comment by maciaszczykm** [comment]
@cheld Done, please take a look.

File: docs/devel/localization.md [hunk 1]
```diff
@@ -61,6 +61,32 @@ const i18n = {
 
 In the HTML code, use one-time bindings like `{{::ctrl.i18n.MSG_EXAMPLE_TEXT}}` for efficiency.
 
+## Localization with `[[Message|]]` pattern
+
+Usage of `MSG_` variables in JavaScript and HTML files is not the only way to localize messages. The second option
+is to use specific pattern directly in HTML messages:
+
+```
+[[This is an example|a simple example]]
```
Comment (cheld) [inline docs/devel/localization.md]: [[This is an example|a simple description for the translator]]

Comment (maciaszczykm) [inline docs/devel/localization.md]: Done, I kept it aligned with example above.

**Pull request merged**